src/whoami_assets/public/WhoamiProcess.ts

```diff
@@ -52,10 +66,19 @@ export default async function WhoamiProcess(
  *     `authenticator.sendAuthenticationRequest` (which will redirect away)
  * * pass the session to `authenticator.useSession`
  * @param this Window
+ * @param options options
+ * @param options.onIdentity - called with each identity
  */
-async function authenticate(this: Pick<typeof globalThis, "document">) {
+async function authenticate(this: Pick<typeof globalThis, "document">, options: {
+  onIdentity(identity: PublicIdentity): void
+}) {
   const log = makeLog("authenticate");
-
+  const authenticator = new Authenticator({
+    identityProvider: {
+      url: new URL('https://auth.ic0.app/authorize')
+    },
+    transport: DefaultAuthenticatorTransport(document),
```

Comment:
If this isn't strictly required, don't specify it.

intention is that transport shouldn't be required even if overriding `identityProvider`